services/src/datasets/external/netcdfcf/ebvportal_provider.rs

```diff
@@ -0,0 +1,555 @@
+
+            groups_list
+                .into_iter()
+                .skip(options.offset as usize)
+                .take(options.limit as usize)
+                .map(|group| {
+                    let mut out_groups = out_groups.clone();
+                    out_groups.push(group.name.clone());
+                    Ok(CollectionItem::Collection(LayerCollectionListing {
+                        id: ProviderLayerCollectionId {
+                            provider_id: EBV_PROVIDER_ID,
+                            collection_id: EbvCollectionId::Group {
+                                class: class.to_string(),
+                                ebv: ebv.to_string(),
+                                dataset: dataset.to_string(),
+                                groups: out_groups,
+                            }
+                            .try_into()?,
+                        },
+                        name: group.title,
+                        description: group.description,
+                        entry_label: None,
+                        properties: None,
+                    }))
+                })
+                .collect()
+        }
+    }
+}
+
+#[async_trait]
+impl LayerCollectionProvider for EbvPortalDataProvider {
+    async fn collection_items(
+        &self,
+        collection: &LayerCollectionId,
+        options: Validated<LayerCollectionListOptions>,
+    ) -> Result<Vec<CollectionItem>> {
+        let id: EbvCollectionId = serde_json::from_str(&collection.0)?;
+
+        let options = options.user_input;
+
+        Ok(match id {
+            EbvCollectionId::Classes => Self::get_classes_collections(&options).await?,
+            EbvCollectionId::Class { class } => {
+                Self::get_class_collections(collection, &class, &options).await?
+            }
+
+            EbvCollectionId::Ebv { class, ebv } => {
+                Self::get_ebv_collections(&class, &ebv, &options).await?
+            }
+            EbvCollectionId::Dataset {
+                class,
+                ebv,
+                dataset,
+            } => {
+                self.get_dataset_collections(&class, &ebv, &dataset, &options)
+                    .await?
+            }
+            EbvCollectionId::Group {
+                class,
+                ebv,
+                dataset,
+                groups,
+            } => {
+                self.get_group_collections(&class, &ebv, &dataset, &groups, &options)
+                    .await?
+            }
+            EbvCollectionId::Entity {
+                class: _,
+                ebv: _,
+                dataset: _,
+                groups: _,
+                entity: _,
+            } => return Err(Error::InvalidLayerCollectionId),
+        })
+    }
+
+    async fn root_collection_id(&self) -> Result<LayerCollectionId> {
+        EbvCollectionId::Classes.try_into()
+    }
+
+    async fn get_layer(&self, id: &LayerId) -> Result<Layer> {
```

Comment:
Why can't you call get_layer from the normal NetCDF provider?

Reply:
because for the ebv portal the path to the netcdf file is retrieved from the dataset via id and in the raw netcdfprovider the path is passed as part of the LayerCollectionId

Reply:
Is it now only working when we have an overview? Because we have to discuss that with idiv.

Reply:
no, this method builds the layer from a `NetCdfOverview` which I thought to be more like a NetCfdSummary and something else than the metdata.json in the overview folder?